mypy/test/data/check-expressions.test

```diff
@@ -1322,8 +1322,10 @@ class str: pass
 def f(x: int) -> None:
     x = yield f('')
     x = 1
+[builtins fixtures/for.py]
 [out]
 main: note: In function "f":
+main:1: error: The return type of a generator function should be "Generator" or one of its supertypes
```

Comment:
I would rather fix up f()'s declaration to define the right return type? (e.g. Iterator[int])